runtime/sema/check_composite_declaration.go

```diff
@@ -468,10 +422,15 @@ func (checker *Checker) declareNestedDeclarations(
 
 			firstNestedCompositeDeclaration := nestedCompositeDeclarations[0]
 
-			reportInvalidNesting(
-				firstNestedCompositeDeclaration.DeclarationKind(),
-				firstNestedCompositeDeclaration.Identifier,
-			)
+			// composite-in-interface nesting errors are already reported elsewhere
```

Comment:
Here and in the similar cases below: Where is it already reported? Could this maybe be unified in one place (here?), so the filtering code here is not necessary? (of course not if this makes the code worse)